app/src/main/java/com/example/makeitso/model/service/impl/StorageServiceImpl.kt

```diff
@@ -17,111 +17,72 @@ limitations under the License.
 package com.example.makeitso.model.service.impl
 
 import com.example.makeitso.model.Task
+import com.example.makeitso.model.service.AccountService
 import com.example.makeitso.model.service.StorageService
-import com.google.firebase.firestore.DocumentChange.Type.REMOVED
-import com.google.firebase.firestore.ListenerRegistration
+import com.example.makeitso.model.service.trace
 import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.firestore.ktx.snapshots
 import com.google.firebase.firestore.ktx.toObject
+import com.google.firebase.firestore.ktx.toObjects
 import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.tasks.asDeferred
+import kotlinx.coroutines.tasks.await
 import javax.inject.Inject
 
-class StorageServiceImpl @Inject constructor() : StorageService {
-    private var listenerRegistration: ListenerRegistration? = null
-
-    override fun addListener(
-        userId: String,
-        onDocumentEvent: (Boolean, Task) -> Unit,
-        onError: (Throwable) -> Unit
-    ) {
-        val query = Firebase.firestore.collection(TASK_COLLECTION).whereEqualTo(USER_ID, userId)
-
-        listenerRegistration = query.addSnapshotListener { value, error ->
-            if (error != null) {
-                onError(error)
-                return@addSnapshotListener
-            }
-
-            value?.documentChanges?.forEach {
-                val wasDocumentDeleted = it.type == REMOVED
-                val task = it.document.toObject<Task>().copy(id = it.document.id)
-                onDocumentEvent(wasDocumentDeleted, task)
-            }
-        }
-    }
+private const val TASK_COLLECTION = "Task"
```

Comment:
Can we put these in a companion object inside the class? (Like we do in `ConfigurationServiceImpl`).